plugin/storage/grpc/shared/archive.go

```diff
@@ -44,7 +44,7 @@ type archiveWriter struct {
 
 // GetTrace takes a traceID and returns a Trace associated with that traceID from Archive Storage
 func (r *archiveReader) GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error) {
-	stream, err := r.client.GetArchiveTrace(upgradeContextWithBearerToken(ctx), &storage_v1.GetTraceRequest{
+	stream, err := r.client.GetArchiveTrace(upgradeWithTraceContext(upgradeContextWithBearerToken(ctx)), &storage_v1.GetTraceRequest{
```

Comment:
I think it might be time to split the statement here, creating the context in one and using it in another.

Reply:
I've used a functional type with a composition function to refactor this part. I suppose it is fine now.